### ISSUE TITLE
Wait, What?

### ISSUE THREAD

**Issue by LutraVibes** [nbarkhina/N64Wasm#38]
What happened here, and how do I fix it?

![image](https://github.com/nbarkhina/N64Wasm/assets/123699770/3b0c0adc-36b1-417f-9c51-ff8a42231894)

**Comment by nbarkhina** [comment]
that means you don't have Emscripten loaded. You need to run this command inside the emsdk folder each time you close and reopen your terminal to get it back into the path

```
source ./emsdk_env.sh
```

**Comment by MajesticWafer** [comment]
I’m having the same problem here. I keep trying to do so but something about Glide64 keeps giving me errors. Honestly I think everybody would benefit from a tutorial video 😆

**Comment by nbarkhina** [comment]
@MajesticWafer what error are you getting in the terminal? I actually did a tutorial on some Emscripten basics several years back:

https://www.youtube.com/watch?v=WHwl8mqVRUg&t=1673s

Not sure if it's all totally applicable. I guess in my readme I assumed folks would have some basic background in emscripten, but  maybe I shouldn't make that assumption.

**Comment by MajesticWafer** [comment]
Funnily enough, I’m getting the same error as @LutraVibes. Then again, I don’t have much (if any) knowledge about emscripten and web assembly. I don’t, however, have much time to watch an hour video, if you know what I mean. 😢

**Comment by nbarkhina** [comment]
@MajesticWafer @LutraVibes another thing you guys can do is create a Codespace in Github. Switch it to the devcontainer branch and click Create Codespace. I have a configuration file in that branch that will automatically give you an environment set up to build N64Wasm. Once it provisions the codespace literally all you have to do is cd into the "code" directory and run "make". it will build and copy the outputed wasm files into the dist folder. See screenshot below with the 2 red bubbles:

![image](https://github.com/nbarkhina/N64Wasm/assets/10982154/71721ddc-e086-42f2-b7c2-46da8d61ef5c)

**Comment by nbarkhina** [comment]
@beezbumba see above this gives you an easy way to build the n64wasm.js and n64wasm.wasm files if you don't have a dev environment set up for building the code.

**Comment by BeezBumba** [comment]
I did some things but I have a error that says:  No such file or directory: '/workspaces/N64Wasm/emsdk/upstream/bin/clang'
But I do have them, they are just grayed out 
![Screenshot 2024-02-10 193633](https://github.com/nbarkhina/N64Wasm/assets/130922034/5abfcd64-b9ab-49b1-935a-b62b111f34c0)
![Screenshot 2024-02-10 193644](https://github.com/nbarkhina/N64Wasm/assets/130922034/aeb6c7b3-3ef5-494d-b8ac-f8130abea0f3)

**Comment by nbarkhina** [comment]
did you use the "devcontainer" branch or the "master" branch when you created the codespace? it looks like your folders in the wrong place. like why is "dist" at the same level as "emsdk" ? if you use the devcontainer branch everything should be set up for you so you don't even need to install emscripten.

**Comment by BeezBumba** [comment]
I think I got confused. I though I had to fork the repository, then make the code space. The fork was all unnecessary.

**Comment by nbarkhina** [comment]
Yeah my codespace was also going into recovery a few times. Usually what I do when that happens is I just delete the codespace and try again after a while. Were you able to finally provision it?

**Comment by BeezBumba** [comment]
Yes I was!

**Comment by nbarkhina** [comment]
ok there is an issue with the devcontainer branch - I'm getting the same clang missing error. I will look into that. In the meantime if you are using a codespace then yeah you will need to install emscripten manually after it provisions using the instructions in the documentation. just make sure to install it in a directory outside of the N64Wasm enlistment.
